lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -351,31 +358,28 @@ prop_putStakePutStake DBLayer {..} epoch a b =
         monitor $ classify (null a && null b) "a & b are empty"
         assert (L.sort res == L.sort b)
 
-prop_stakePoolOwner
+-- | Heavily relies on the fact that PoolId have a entropy that is sufficient
+prop_poolRegistration
     :: DBLayer IO
-    -> StakePoolOwnersFixture
+    -> [PoolRegistrationCertificate]
     -> Property
-prop_stakePoolOwner DBLayer {..} (StakePoolOwnersFixture poolId entries) =
+prop_poolRegistration DBLayer {..} entries =
```

Comment:
**[remark]**

As we have now  PoolMetadata decoupled from PoolOwner it would be nice to introduce properties like : 
1. registering pools and see if we ONLY have pool metadata relating to this
2. registering pools and then we have pool metadata not belonging to the pool we registered - and what happens?
3. registering and deregistring some pools and what about pool metadata consistency (maybe we cannot deregister pool)

Reply:
We can't retire pool at the moment indeed, we can only register new pools indeed. The property currently test that for any arbitrary list of registration, if you insert them, you can read them back, therefore making sure that the binding between owners and certificate is correctly respected. It covers your second case (where many metadata does not belong to your pool, but still you don't get extra metadata when fetching) and your first case as well (when fetching, you only get your metadata).